From e84e9dc582797b5e586ea4ff3ae7b5f3656968ee Mon Sep 17 00:00:00 2001
From: Alessandro Astone <ales.astone@gmail.com>
Date: Sun, 1 Mar 2026 18:14:09 +0100
Subject: [PATCH] gallivm: Fix armhf build against LLVM 22

StringMapIterator<bool> became StringMapIterBase<bool, false /* IsConst */>;
Use `auto` to handle either case.

Reviewed-by: Icenowy Zheng <zhengxingda@iscas.ac.cn>
Part-of: <https://gitlab.freedesktop.org/mesa/mesa/-/merge_requests/40161>
---
 src/gallium/auxiliary/gallivm/lp_bld_misc.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git ./src/gallium/auxiliary/gallivm/lp_bld_misc.cpp ./src/gallium/auxiliary/gallivm/lp_bld_misc.cpp
index ce8ca02bf74..86779f013c7 100644
--- ./src/gallium/auxiliary/gallivm/lp_bld_misc.cpp
+++ ./src/gallium/auxiliary/gallivm/lp_bld_misc.cpp
@@ -331,7 +331,7 @@ lp_build_fill_mattrs(std::vector<std::string> &MAttrs)
       llvm::sys::getHostCPUFeatures(features);
    #endif
 
-   for (llvm::StringMapIterator<bool> f = features.begin();
+   for (auto f = features.begin();
         f != features.end();
         ++f) {
       MAttrs.push_back(((*f).second ? "+" : "-") + (*f).first().str());
-- 
2.54.0

